pdr_backend/pdr_dashboard/util/helpers.py

```diff
@@ -90,4 +154,69 @@ def get_date_period_text_for_selected_predictoors(payouts: pl.DataFrame) -> str:
 
 @enforce_types
 def get_date_period_text_header(start_date: UnixTimeS, end_date: UnixTimeS) -> str:
-    return _format_date_text(start_date.to_dt(), end_date.to_dt(), True)
+    return _format_date_text(start_date.to_dt(), end_date.to_dt())
+
+
+def get_empty_feeds_filters() -> List[Filter]:
+    return [Filter(**item) for item in EMPTY_FEEDS_FILTERS]
+
+
+@enforce_types
+def produce_feeds_filter_options(
+    df: pl.DataFrame,
+) -> Tuple[List[str], List[str], List[str], List[str]]:
+    df = df.with_columns(
+        pl.col("pair")
+        .str.split_exact("/", 1)
+        .map_elements(lambda x: x["field_0"], return_dtype=pl.String)
+        .alias("base_token"),
+        pl.col("pair")
+        .str.split_exact("/", 1)
+        .map_elements(lambda x: x["field_1"], return_dtype=pl.String)
+        .alias("quote_token"),
+        pl.col("source").str.to_titlecase().alias("source"),
+    )
+
+    return (
+        df["base_token"].unique().to_list(),
+        df["quote_token"].unique().to_list(),
+        df["source"].unique().to_list(),
+        df["timeframe"].unique().to_list(),
+    )
+
+
+def check_data_loaded(
```

Comment:
I'm also wondering whether this decorator can help us with the sorting part which we eliminated as an option at some point

Reply:
maybe we can add another decorator for this solution